android/src/main/java/com/mattermost/networkclient/APIClientModule.kt

```diff
@@ -122,10 +127,56 @@ class APIClientModule(reactContext: ReactApplicationContext) : ReactContextBaseJ
         }
     }
 
+    @ReactMethod
+    fun upload(baseUrl: String, endpoint: String, file: String, taskId: String, options: ReadableMap?, promise: Promise) {
+        val skipBytes = if (options != null && options.hasKey("skipBytes")) options.getInt("skipBytes").toLong() else 0;
+        val stream = if (options != null && options.hasKey("stream")) options.getBoolean("stream") else false;
+
+        try {
+            // Create a Request Body
+            val body: RequestBody = if (stream) {
+                UploadFileRequestBody(reactApplicationContext, Uri.parse(file), skipBytes, ProgressListener(reactApplicationContext, taskId))
+            } else {
+                MultipartBody.Builder().addPart(UploadFileRequestBody(reactApplicationContext, Uri.parse(file), skipBytes, ProgressListener(reactApplicationContext, taskId))).build()
+            }
+
+            // Create a Request
+            var request = sessionsRequest[baseUrl]!!.url("$baseUrl/$endpoint").post(body);
```

Comment:
Can't the method be put and or patch? Is it true that this must be a post method at all times?

Reply:
Yup it can be! Should we allow for any of the three and just leave `post` as default?

Reply:
Yup, we should allow the method to be defined and fallback to post if undefined